### DISCUSSION TITLE
How to obtain the action sequence for each round after using a custom environment

### DISCUSSION THREAD

**Discussion by jinnanchu** [PKU-Alignment/omnisafe#337] [Q&A]
I have already created my own environment and can use algorithms for training, but the training results do not include the specific sequence of selected actions for each round. How should this be displayed? Thank you

**Comment by Gaiejj** [comment]
Omnisafe currently does not support step-wise recording as it would result in excessive storage overhead. You may try modifying the adapter.py file of the corresponding algorithm to manually save the step-wise action sequence as txt/csv/json files.
